### PR TITLE
Fix wasm build ci

### PR DESCRIPTION
**Pull request by mcbarton**
This PR will fix the broken wasm builds in the ci
Fixes https://github.com/compiler-research/CppInterOp/issues/294

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
 cmake_minimum_required(VERSION 3.13)
+set(CppInterOp_Version "1.4.0;dev")
```
Comment (vgvassilev) [inline CMakeLists.txt:2]: We still need to read that from the file.

**Comment by maximusron** [comment]
@mcbarton we have some failures on main that seem to be happening even after reverting commits so I was wondering if clearing the cache might resolve it. would now be a good time or would you like to push more commits?

**Comment by mcbarton** [comment]
@maximusron This PR is not ready. Rebuild the cache and I will revisit this PR to fix the wasm jobs.

**Comment by vgvassilev** [comment]
Are we keeping the debug statements?

**Comment by mcbarton** [comment]
> Are we keeping the debug statements?

The debug statements are not staying. They are just there because the PR is not currently working. I'm trying to work out why ;dev is still in the version name in the config file.

**Comment by mcbarton** [comment]
@vgvassilev The debug run says this is doing what is expected. The ;dev in the version number in the config file is coming from somewhere else. Any ideas where?

**Comment by vgvassilev** [comment]
Here: https://github.com/compiler-research/CppInterOp/blob/main/cmake/CppInterOp/CppInterOpConfigVersion.cmake.in ?

File: cmake/CppInterOp/CppInterOpConfigVersion.cmake.in [hunk 2]
```diff
@@ -1,4 +1,4 @@
-set(PACKAGE_VERSION "@CPPINTEROP_VERSION@")
+set(PACKAGE_VERSION "@CPPINTEROP_VERSION_MAJOR@.@CPPINTEROP_VERSION_MINOR@.@CPPINTEROP_VERSION_PATCH@")
```
Comment (vgvassilev) [inline cmake/CppInterOp/CppInterOpConfigVersion.cmake.in:1]: We need to signal if that came from a dev version or not somehow.

Comment (mcbarton) [reply cmake/CppInterOp/CppInterOpConfigVersion.cmake.in:1]: I could have a variable contain the last thing in the version list. If it is the word dev then we could output a message to the screen telling the user that they are working with a dev version.

Comment (vgvassilev) [reply cmake/CppInterOp/CppInterOpConfigVersion.cmake.in:1]: On a second thought let's ignore this for now. Maybe we can keep track of this problem in a separate issue.

**Comment by maximusron** [comment]
@mcbarton looks like this should be good to go now. Let's rebase with main before the last checks are done just to be sure

**Comment by mcbarton** [comment]
> @mcbarton looks like this should be good to go now(based on the passing wasm builds). Let's rebase with main before the last checks are done just to be sure

@maximusron I think it is ok too. I've just done a cleanup of things I think are unneeded changes. Once the changes pass I will rebase.

**Comment by maximusron** [comment]
sounds good

**Comment by maximusron** [comment]
I'm going to cancel the run on main to speed this up

File: cmake/CppInterOp/CppInterOpConfigVersion.cmake.in (hunk 2 above)
Comment (mcbarton) [reply cmake/CppInterOp/CppInterOpConfigVersion.cmake.in]: We have managed to retain the ;dev part in package version in the end.

**Comment by maximusron** [comment]
@mcbarton can you rebase into a single commit before the checks?

**Comment by mcbarton** [comment]
> @mcbarton can you rebase into a single commit before the checks?

I will rebase it in around 1-2 hours time. I am away from my computer until then.

**Review by maximusron — APPROVED** [review]

**Pull request merged**